src/modules/problemMethods.ts

```diff
@@ -266,14 +272,42 @@ const getList = async (
     return { nextCursor, items: secondQueryItems, total };
 };
 
-const update = (data: UpdateProblem): Promise<Problem> => {
+const update = async (data: UpdateProblem, authorId: number): Promise<Problem> => {
+    const problem = await prisma.problem.findFirstOrThrow({
+        where: { id: data.problemId },
+    });
+
     const { problemId, tagIds, ...restData } = data;
     const updateData: Prisma.ProblemUpdateInput = { ...restData };
 
     if (tagIds && tagIds.length > 0) {
         updateData.tags = { set: idsToIdObjs(tagIds) };
     }
 
+    if (data.description !== problem.description) {
+        await prisma.problemHistory.create({
+            data: {
+                subject: 'description',
+                user: { connect: { id: authorId } },
+                problem: { connect: { id: data.problemId } },
+                previousValue: problem.description,
+                nextValue: data.description,
+            },
+        });
+    }
+
+    if (data.solution !== problem.solution) {
```

Comment:
should be a single transaction with problem.update (since it's useless if update fails)